common/web/types/src/consts/virtual-key-constants.ts

```diff
@@ -159,6 +159,19 @@ export const USVirtualKeyMap: KeyMap = [
   [ k.K_SPACE ],
 ];
 
+export const KSVirtualKeyMap: KeyMap = [
+  // ` 1 2 3 4 5 6 7 8 9 0 - = [bksp]
+  [ k.K_BKQUOTE, k.K_1, k.K_2, k.K_3, k.K_4, k.K_5, k.K_6, k.K_7, k.K_8, k.K_9, k.K_0, k.K_HYPHEN, k.K_EQUAL, k.K_BKSLASH /* YEN */  ],
```

Comment:
> On a Japanese keyboard, 0x7D produces `K_BKSLASH`, and 0x2B produces `K_oC1`. That may be the safest pattern for us to match.

And then I realised that that won't work. 0x2B is `K_BKSLASH` on kbdus. So we'll need to use something else for 0x7D (`K_oC1`?) in order to differentiate, because we can't change the already assigned 0x2B. The actual code doesn't really matter all that much so long as we are consistent internally.

Reply:
So I am wondering about the advisability of these layout maps, as they form a separate source of overlapping truth -- we already have scanCodes-implied.xml. I think we should just a have single scancode:vk map somewhere, and construct layout maps from that+scanCodes-implied.xml.

Reply:
could it be `k.K_oE2` ? which is sometimes backslash-like things in ISO and ABNT2

Reply:
@mcdurdin  scancodes to vkeys… seems to make sense, should I just build up a table from the data we already have here? (maybe merge this first as a baseline)

On the maybe-plus side, that means we actually could support custom scancodes! Did I say that out loud?

Reply:
`K_oE2` is mapped to scan code 0x56. I don't think we should use the same VK for different scan codes.

Internally we probably could have been using scan codes directly rather than VKs. But that boat has sailed.

Reply:
okay.  I'll keep looking.

it sounds like there should also be a test,  a sort of virtual-`@mcdurdin`, which verifies 1:1 mapping from scan code to vkey here.

Reply:
that's now given by definition…